extensions/2.0/Khronos/KHR_interactivity/Specification.adoc

```diff
@@ -0,0 +1,4196 @@
+
+=== Turing Completeness
+The execution model and node choices for this extension mean that it is Turing-complete. This means that an implementation of this can execute any computation and it is also hard to predict if it will run forever, e.g., halt or not.
+
+While this may present security implications, it is not a major hindrance and can be safely mitigated so that any implementation does not become susceptible to denial of services by badly behaving behavior graphs, whether intentional or not.
+
+The main way to mitigate the risk of non-halting behavior graphs is to limit the amount of time given to them for execution, both in terms of individual time slice as well as overall execution time.
+
+
+[[concepts]]
+= Concepts
+
+[[concepts-general]]
+== Graphs
+
+A behavior graph is a JSON object containing _nodes_. It **MAY** also contain custom variables and custom events.
+
+Behavior graphs are directed graphs with no directed cycles.
+
+When a glTF asset contains a behavior graph, all glTF animations are assumed to be controlled by the graph so they **MUST NOT** play automatically.
+
+[[nodes]]
+== Nodes
+
+A _node_ is a JSON object, which represents an executable item. Each node is defined by its _declaration_, which includes an _operation_ and a (possibly empty) set of _value sockets_. Node operations follow `domain/operation` naming pattern. Depending on the operation, a node **MAY** have input and/or output _flow sockets_; they **MAY** be affected by the node's _configuration_.
+
+[[nodes-operation]]
+=== Operation
+
+A node's _operation_ defines a specific set of steps performed by the execution environment when the node is executed.
+
+A node is executed when one of its input flow sockets is activated, when one of its output value sockets is accessed by another node, when an operation-specific event occurs. A node **MAY** repeatedly activate its own input flow sockets during the execution.
+
+Usually, the node execution includes evaluating its input value sockets (if any), processing its own logic, and activating any number (including zero) of output flow sockets.
+
+[[nodes-sockets]]
+=== Sockets
+
+There are four kinds of sockets.
+
+_Output value sockets_ represent data initialized by the node or produced during its execution. For example, it could be results of math operations or parts of the node's internal state. Accessing these sockets either triggers computing the return value on the fly by executing the node or returns a value based on the node's internal state. Exact behavior depends on the node's operation. As a general rule, output value sockets **MUST** retain their values until a node having any number of flow sockets is executed.
```

Comment:
The last sentence is a bit hard to understand. Would it be right to say "until a node with any number of flow sockets is executed." ?

Reply:
Done.